### ISSUE TITLE
Don't require desktop development workload for Windows

### ISSUE THREAD

**Issue by adsamcik** [flutter/flutter#58978]
After latest master version upgrade I have started seeing:

```
Visual Studio is missing necessary components. Please re-run the Visual Studio installer for the "Desktop development with C++" workload, and include these components:
        MSVC v142 - VS 2019 C++ x64/x86 build tools
         - If there are multiple build tool versions available, install the latest
        Windows 10 SDK
```

It seems the check now relies on "Desktop development with C++". If it is not installed the flutter won't build even when one has all the required components (previously it built just fine), this started ocurring after recent flutter upgrade. After installing the whole Desktop development package the issue disappeared (MSVC was already installed). I don't think it's good idea to force everyone to install everything since on systems with smaller disk space it can waste precious space.

# Reproduction

1. Install required VS components (without installing Desktop development with C++)
2. Run flutter doctor (or build)

**Comment by stuartmorgan** [comment]
Fixing title to accurately describe what's being requested; this isn't a failure to detect the toolchain component. It's correctly determining that you don't have the workload that the documentation indicates you need.

> It seems the check now relies on "Desktop development with C++".

This is correct.

> If it is not installed the flutter won't build even when one has all the required components (previously it built just fine)

The instructions have always said to install that workload; the difference is that we're now actually checking. "Required components" is a somewhat ambiguous term here since it could mean either what our checks require, or what would technically be enough to successfully build. By the former definition, you don't have all the required components. The distinction is important because I'm almost certain there were components that weren't checked before, but were actually needed in order to build and run (e.g., the redistributables that are needed by the runtime).

> I don't think it's good idea to force everyone to install everything since on systems with smaller disk space it can waste precious space.

That workload is a small fraction of everything in the VS installer; its list of required components is quite short.

We've had more issues and questions about correctly setting up VS than probably any other single topic across all of the desktop platforms; people have made instructional videos just showing people how to find and check the correct checkboxes. The recent change was specifically made to simplify the setup process to address this longstanding issue. Micro-optimizing for size (especially given that the recommended instructions already said to do this, so in practice will not affect the resulting install size for the vast majority of users) is much less important than ease of use.


If someone wants to figure out exactly the set of necessary components and a robust, future-proof (i.e., not re-introducing the SDK version problem) approach for checking for exactly those components, and it doesn't increase the complexity of the basic instructions, I will review it, but it's not something I expect the Flutter team to ever work on. (And if someone does do that, please include an actual number for how much space is saved, so that can be weighed against the added code complexity.)

**Comment by adsamcik** [comment]
Ok from what you described I can see why it would be preferable to just simply check for the workload instead of each component specifically and taking into account future-proofness, it really would be a waste of time to work on this.

Previously I had custom setup from previous work with C++ so I never noticed the message in earlier builds since I had all the components it checked for and everything worked. 

Installing the workload took roughly 1gb of extra space.

**Comment by stuartmorgan** [comment]
> Installing the workload took roughly 1gb of extra space.

Interesting, that's much larger than I would have guessed. I wonder which components are the bulk of that.

**Comment by adsamcik** [comment]
I inspected it again and I really don't understand what could've taken so much space. I noticed that around 300-400MB can be reduced by unchecking all the non-essential optional components. But I really couldn't find what would take the extra space. It is possible that something extra was installed that I couldn't find though.

From what I could see, the workload should not contain much extra than what is needed by flutter.

After going through all this in more detail, if you agree I think this issue can be closed, since there is not much value in this and the 1gb - the non-essential optionals could have simply been a space approximation error.
